static/srcView.js

```diff
@@ -178,17 +179,69 @@ export class SourceView extends React.Component {
             refMenu = <RefMenu app={this.props.app} location={this.state.refMenu} onClose={onClose} target={this.state.refMenu.target} id={this.state.refMenu.id} />;
         }
 
+        const allowedTags = [
+            'h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'blockquote', 'p', 'a', 'ul', 'ol', 'nl', 'li', 'b', 'i',
+            'img', 'strong', 'em', 'strike', 'code', 'hr', 'br', 'div', 'table', 'thead', 'caption', 'tbody',
+            'tr', 'th', 'td', 'pre',
+        ];
+
+        const View = {
+            RENDERED: 'content',
+            SOURCE: 'source',
+        };
+
+        let viewSelector;
+        let currentView = this.state.currentView;
+
+        const setView = (to) => {
+            this.setState({ currentView: to })
+        };
+
+        switch (true) {
```

Comment:
This is a pretty weird looking idiom. Is it common in JS? Is it much better than if/else?

Reply:
It's a pretty well-known idiom but it's debated whether it's more readable than `if`. I think `if` would be better here - less weird and wouldn't require `!!`